Sources/HaCWebsiteLib/Events/Event.swift

```diff
@@ -0,0 +1,14 @@
+import Foundation
+
+protocol Event {
+    var title : String {get}
+    var time : Date {get}
+    var eventDescription: String {get} //NOT related to event content (e.g. workshop content)
+    var color : String {get}
+    var hypePeriod : DateInterval {get}
+    var imageURL : String? {get}
+    var endTime : Date? {get}
+    var venue : String? {get} //for room names etc.
```

Comment:
Did we want to have a location as coordinates here? We could encode this like

```
{
  coordinates: {
    latitude: Double;
    longitude: Double;
  }
  address?: String;
  venue?: String;
}

Reply:
Yeah, good idea - do we think this will solely be used for Events? I.e. should I put it in the same directory as the rest of the events stuff?

Reply:
I think this will be a useful struct for general use (for example rendering a location)